File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Java/models/DeliveryPerson.java
package models;

import utils.DeliveryStatus;

public class DeliveryPerson {
    private String id;
    private String name;
    private String phone;
    private DeliveryStatus status;

    public DeliveryPerson(String id, String name, String phone, DeliveryStatus status) {
        this.id = id;
        this.name = name;
        this.phone = phone;
        this.status = status;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getPhone() {
        return phone;
    }

    public DeliveryStatus getStatus() {
        return status;
    }
}

File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Cpp/models/src/DeliveryPerson.cpp
#include "../include/DeliveryPerson.h"

DeliveryPerson::DeliveryPerson(const std::string& id,
                               const std::string& name,
                               const std::string& phone,
                               DeliveryStatus status)
    : id(id), name(name), phone(phone), status(status) {}

const std::string& DeliveryPerson::getId() const {
    return id;
}

const std::string& DeliveryPerson::getName() const {
    return name;
}

const std::string& DeliveryPerson::getPhone() const {
    return phone;
}

DeliveryStatus DeliveryPerson::getStatus() const {
    return status;
}

void DeliveryPerson::setStatus(DeliveryStatus newStatus) {
    status = newStatus;
}
